server/services/caseListService.js

```diff
@@ -30,17 +41,12 @@ module.exports = function createCaseListService(nomisClientBuilder, roService, l
   /**
    * Assume username is assigned to an RO.  Look up this user in the local db (staff_ids table) and take the staff code.
    * Alternatively if a unique staff code cannot be found in this db, ask Delius for a staff code.
-   *
-   * The code below is an ( ugly bodge / elegant solution ) that breaks the type contract on Result.orElse():
-   * staffCodeFromDB is a Result<string,string>, but getStaffCodeFromDelius() returns a Promise<Result<string,string>>
-   * This means that the .orElse() will return either a Result<string,string> or a Promise<Result<string,string>>
-   * Fortunately the 'await' prefix flattens nested Promises so that it all 'just works'.
-   *
    */
   const getROCaseList = (username, token) => async () => {
     const staffCodeFromDb = await getStaffCodeFromDb(username)
-    const staffCode = await staffCodeFromDb.orElse(() => getStaffCodeFromDelius(username))
-    return staffCode.map(getOffendersForStaffCode(token)).match(R.identity, message => ({ hdcEligible: [], message }))
+    const staffCode = await staffCodeFromDb.orElseTryAsync(() => getStaffCodeFromDelius(username))
+    const offendersForStaffCode = await staffCode.mapAsync(getOffendersForStaffCode(token))
+    return offendersForStaffCode.match(R.identity, message => ({ hdcEligible: [], message }))
```

Comment:
The async methods produce `Promise<Result<S,E>>` which will tend to break any method chaining. 

I guess there might be a way of keeping the `Promise` inside the `Result` but I suspect this would be much more  complicated.